### ISSUE TITLE
PR checks should require tests like they require docs

### ISSUE THREAD

**Issue by iritkatriel** [python/core-workflow#480]
tests are as standard a requirement as doc changes.  New PR authors who are not aware of this could be helped by a check failing due to lack of tests (rather than waiting for someone to comment on that, which can take years in some cases).

So the suggestion is:

Any PR that touches code must also touch the tests, or have a skip-tests label.

**Comment by ezio-melotti** [comment]
It might be better to move this to python/bedevere, since he is the one performing checks related to the `skip-*` labels.

**Comment by terryjreedy** [comment]
AFAIK, there is no doc check/requirement, but I think there should be for enhancements.  As I said on discord, I think we should at least try out both.

**Comment by hugovk** [comment]
> It might be better to move this to python/bedevere, since he is the one performing checks related to the `skip-*` labels.

At the sprint we talked about replacing some Bedevere functionality with more "native" GitHub stuff.

For example, this could be done with GitHub Actions.

@brettcannon Do you think https://github.com/brettcannon/check-for-changed-files would be suitable for this?

**Comment by brettcannon** [comment]
> Do you think https://github.com/brettcannon/check-for-changed-files would be suitable for this?

Yes because that's what I designed it for. 😁 See https://github.com/microsoft/vscode-python/blob/133a8e877aa3755eaf101f1f1edf446432770b40/.github/workflows/pr-file-check.yml#L34-L43 as an example usage for this exact use case.

**Comment by AA-Turner** [comment]
Closing, stale.

**Comment by hugovk** [comment]
Let's re-open this one.

**Comment by AA-Turner** [comment]
I'm not sure how useful it would be in reality. A 'random' sampling of the most recent 25 commits shows that only 8 needed to touch the tests. Of those 8, 1 was a pure refactor and 2 *only* touched the tests. I think a 'skip-tests' label would be more noise that we manually need to apply for lots of PRs.

We could add a note about having tests for new 'feature' work in either the PR template or CONTRIBUTING.rst, neither of which mention tests at the moment.

A
